mitype/app.py

```diff
@@ -447,6 +457,23 @@ def print_stats(self, win):
             curses.color_pair(5),
         )
 
+    def switch_text(self, win, value):
+        """Load next or previous text snippet from database.
+
+        Args:
+            win (any): Curses window.
+            value (int): value to increase or decrement by.
+        """
+        self.text_id += value
```

Comment:
When mitype is started as `mitype -f filename` the text_id variable is a string with the value of `filename` and pressing the arrow keys would crash the app. This case needs to be handled and can be simply achieved here by checking if `text_id` is string, in which case just exit the function and do nothing.

Reply:
Okay, but breaking the function here results in a blank screen.
I think something like this outside the function would work better:
```
if isinstance(self.text_id, str):
    pass
else:
    if is_left_arrow_key(key) and not self.first_key_pressed:
        win.clear()
        self.switch_text(win, -1)

    if is_right_arrow_key(key) and not self.first_key_pressed:
        win.clear()
        self.switch_text(win, 1)
```

What do you think?

Reply:
The screen goes blank because of the lines that clear the screen, move that `win.clear()` call inside the `switch_text` function. That should allow you to return from the function in between as intended.